talisker/util.py

```diff
@@ -35,3 +38,60 @@ def parse_url(url, proto='http'):
 
 def pkg_is_installed(name):
     return name in [x.project_name for x in pip.get_installed_distributions()]
+
+
+# a module level cache for global objects
+_global_cache = {}
+_global_dicts = []
+_context_locals = []
+
+
+def module_cache(func):
```

Comment:
Hmm, I wonder if theres a different dict api I can use. I reluctant to lock the critical section, as that would be hit every single lookup